### ISSUE TITLE
Expose underlying cache or some other means of explicitly clearing the cache

### ISSUE THREAD

**Issue by dkichler** [kuitos/axios-extensions#11]
**Feature Request**

It would be useful to allow clients to explicitly clear the cache in cases where data has become stale.  This could be easily achieved by exposing the cache on the wrapped adapter, but would change the API slightly.  

e.g.
```
const cache = new LRUCache({ maxAge: cacheAge });

return {
  cache,
  adapter: config => {

		const { url, method, params } = config;
...
```
usage:
```
const cacheAdapter = cacheAdapterEnhancer(axios.defaults.adapter, true)
const http = axios.create({
	baseURL: '/',
	headers: { 'Cache-Control': 'no-cache' },
	adapter: cacheAdapter.adapter
});
...
cacheAdapter.cache.reset()
```

**Comment by kuitos** [comment]
@dkichler Thanks for your pr, I had merged it and made a bit refactoring😀. See https://github.com/kuitos/axios-extensions/commit/6134b7ae38afce6985b260be87cfe9d07d3a6a69
and https://github.com/kuitos/axios-extensions/#more-advanced

Now there are two ways that can meet your requirement.
1. define a default cache manually when constructing
```js
const defaultCache = new LRUCache({ maxAge: cacheAge });
const http = axios.create({
	baseURL: '/',
	headers: { 'Cache-Control': 'no-cache' },
	adapter: throttleAdapterEnhancer(cacheAdapterEnhancer(axios.defaults.adapter, true, 'cache',  defaultCache))
});
```
2. specify the cache instance other than the default one when request made
```js
const cache = new LRUCache({ maxAge: cacheAge });
http.get('/users', { cache });
```

**Comment by kuitos** [comment]
released v2.0.0

**Comment by dkichler** [comment]
Very nice, your improvement is far cleaner - does not break the existing API, and allows explicit control over the cache through dependency injection.
